### PR TITLE
Forbid specify database in MariaDB connection url

### PR DESCRIPTION
**Pull request by zhaoyim**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?
fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)
MariaDB connector

> How would you describe this change to a non-technical end user or system administrator?
Forbid specify database in MariaDB connection url

## Related issues, pull requests, and links
fixes [12712](https://github.com/trinodb/trino/issues/12712)

## Documentation

(X) No documentation is needed.

## Release notes

(x) Release notes entries required with the following suggested text:

```markdown
# MariaDB
* Disallow having a database in the `connection-url` config property. ({issue}`12712`)
```

### CODE REVIEW

File: plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbJdbcConfig.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.sql.SQLException;
+
+public class MariaDbJdbcConfig
+        extends BaseJdbcConfig
+{
+    @AssertTrue(message = "Invalid JDBC URL for MariaDB connector")
+    public boolean isUrlValid()
+    {
+        Driver driver = new Driver();
+        return driver.acceptsURL(getConnectionUrl());
```
Comment (ebyhr) [inline plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbJdbcConfig.java:31]: > Forbid specify database in MariaDB connection url

The commit contains other change. Could you separate a commit?

File: plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbJdbcConfig.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @AssertTrue(message = "Database (catalog) must not be specified in JDBC URL for MariaDB connector")
+    public boolean isUrlWithoutDatabase()
+    {
+        try {
+            Configuration conf = Configuration.parse(getConnectionUrl());
+            return conf.database() == null ? true : false;
```
Comment (ebyhr) [inline plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbJdbcConfig.java:39]:
```suggestion
            return conf.database() == null;
```

**Review by ebyhr — APPROVED** [review]

File: plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbJdbcConfig.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestMariaDbJdbcConfig
+{
+    @Test
+    public void testIsUrlWithoutDatabase()
+    {
+        assertTrue(isUrlWithoutDatabase("jdbc:mariadb://example.net:3306"));
+        assertTrue(isUrlWithoutDatabase("jdbc:mariadb://example.net:3306/"));
```
Comment (ebyhr) [inline plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbJdbcConfig.java:27]: Is there any reason to allow an empty character as valid URL?

Comment (zhaoyim) [reply plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbJdbcConfig.java:27]: Thanks for review! Here, I think an empty character is the same as the not specified the database, this most an habit, just like web url, xx.xxx.xx is the same as xx.xxx.xx/ . Also the maria client treat them as the same all return database is null.

File: plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbJdbcConfig.java (hunk 1 above)
Comment (zhaoyim) [reply plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbJdbcConfig.java:31]: Sure! Done. Thanks for review!

File: plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbJdbcConfig.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestMariaDbJdbcConfig
+{
+    @Test
+    public void testIsUrlWithoutDatabase()
```
[the remark this answers is not in the document]
Comment (zhaoyim) [reply plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbJdbcConfig.java]: Sure, updated. Thanks!

File: plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbJdbcConfig.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ * limitations under the License.
+ */
+package io.trino.plugin.mariadb;
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestMariaDbJdbcConfig
```
Comment (ebyhr) [inline plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbJdbcConfig.java:21]: Could you add a test case in `TestMariaDbPlugin` too? You can refer to `TestMySqlPlugin`.

Comment (zhaoyim) [reply plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbJdbcConfig.java:21]: Sure! Added the test.

**Pull request merged**